### ISSUE TITLE
LoginForm.tsx: `OnChange` handler in email_address login form is randomly hanging the program

### ISSUE THREAD

**Issue by recolic** [deltachat/deltachat-desktop#2506]
**At this codeline**: https://github.com/deltachat/deltachat-desktop/blob/69c813ee154ea275194e9bcdddb2bc1ade8229a0/src/renderer/components/LoginForm.tsx#L187

- Operating System (Linux/Mac/Windows/iOS/Android): Linux
- Delta Chat Version: Version 1.26.0 (git: tags/v1.26.0)
- Actual behavior:

While adding an account, while typing in your email address, the program will hang randomly. 

- Steps to reproduce the problem:

1. Install DeltaChat on ArchLinux. 
2. Start deltaChat, Add an account. 
3. Type your email address. For example: `root@recolic.net`
4. On the moment you type `@`, the program is not responding anymore. You need to wait for about **20 seconds**, before the program could recover. 

You can also try to type some random string, or backspace. The program will randomly hang. 

- Screenshots:

![image](https://user-images.githubusercontent.com/19610256/149266874-70954a7a-1f93-4214-bd65-0c5a3b13e954.png)

- Logs:

This is strace log. However, there's no valuable information at all. **It's not blocked by a system call.** 

```
fcntl(255, F_SETFD, FD_CLOEXEC)         = 0
fcntl(255, F_GETFL)                     = 0x8000 (flags O_RDONLY|O_LARGEFILE)
newfstatat(255, "", {st_mode=S_IFREG|0755, st_size=64, ...}, AT_EMPTY_PATH) = 0
lseek(255, 0, SEEK_CUR)                 = 0
read(255, "#!/bin/bash\n\ncd /opt/DeltaChat/e"..., 64) = 64
newfstatat(AT_FDCWD, "/opt", {st_mode=S_IFDIR|0755, st_size=272, ...}, 0) = 0
newfstatat(AT_FDCWD, "/opt/DeltaChat", {st_mode=S_IFDIR|0755, st_size=42, ...}, 0) = 0
newfstatat(AT_FDCWD, "/opt/DeltaChat/electron_app", {st_mode=S_IFDIR|0755, st_size=148, ...}, 0) = 0
chdir("/opt/DeltaChat/electron_app")    = 0
newfstatat(AT_FDCWD, ".", {st_mode=S_IFDIR|0755, st_size=148, ...}, 0) = 0
newfstatat(AT_FDCWD, "/usr/mybin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/home/recolic/go/bin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/mymsbin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/mybin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/home/recolic/go/bin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/mymsbin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/mybin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/home/recolic/go/bin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/mymsbin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/mybin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/home/recolic/go/bin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/mymsbin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/local/bin/electron", 0x7ee8a6e4b5e0, 0) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/bin/electron", {st_mode=S_IFREG|0755, st_size=379, ...}, 0) = 0
newfstatat(AT_FDCWD, "/usr/bin/electron", {st_mode=S_IFREG|0755, st_size=379, ...}, 0) = 0
geteuid()                               = 1000
getegid()                               = 1000
getuid()                                = 1000
getgid()                                = 1000
access("/usr/bin/electron", X_OK)       = 0
newfstatat(AT_FDCWD, "/usr/bin/electron", {st_mode=S_IFREG|0755, st_size=379, ...}, 0) = 0
geteuid()                               = 1000
getegid()                               = 1000
getuid()                                = 1000
getgid()                                = 1000
access("/usr/bin/electron", R_OK)       = 0
newfstatat(AT_FDCWD, "/usr/bin/electron", {st_mode=S_IFREG|0755, st_size=379, ...}, 0) = 0
newfstatat(AT_FDCWD, "/usr/bin/electron", {st_mode=S_IFREG|0755, st_size=379, ...}, 0) = 0
geteuid()                               = 1000
getegid()                               = 1000
getuid()                                = 1000
getgid()                                = 1000
access("/usr/bin/electron", X_OK)       = 0
newfstatat(AT_FDCWD, "/usr/bin/electron", {st_mode=S_IFREG|0755, st_size=379, ...}, 0) = 0
geteuid()                               = 1000
getegid()                               = 1000
getuid()                                = 1000
getgid()                                = 1000
access("/usr/bin/electron", R_OK)       = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigprocmask(SIG_BLOCK, [INT TERM CHLD], [], 8) = 0
lseek(255, -1, SEEK_CUR)                = 63
clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x62fc6e44bed0) = 1400147
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
rt_sigaction(SIGINT, {sa_handler=0xba80bbbab00, sa_mask=[], sa_flags=SA_RESTORER, sa_restorer=0x62fc6e4fcda0}, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=SA_RESTORER, sa_restorer=0x62fc6e4fcda0}, 8) = 0
wait4(-1, Logfile: /home/recolic/.config/DeltaChat/logs/2022-01-13-12-31-38.log
[1400147:0113/123138.321486:ERROR:cert_issuer_source_aia.cc(32)] Error parsing cert retrieved from AIA (as DER):
ERROR: Couldn't read tbsCertificate as SEQUENCE
ERROR: Failed parsing Certificate

[1400147:0113/123138.323550:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for 0.0.0.0 failed:
----- Certificate i=0 (CN=recolicmspc.mscorp.rtmp.asia) -----
ERROR: Time is after notAfter


init: 807.939ms

```

**Comment by recolic** [comment]
There is a known workaround for this issue: 

1. Type your email address in notepad. 
2. Copy-Paste it to delta chat.

**Comment by Jikstra** [comment]
It's possible that under certain conditions the core call hangs and blocks the whole backend and renderer process (electron sadly does it this way). However I cannot reproduce this on my computer, but i know that there were problems with this before. Can you maybe attach a of the core events? You can start deltachat with debug logs this way: `deltachat --devmode`

**Comment by recolic** [comment]
I have got the following debug logs: 

<details>
  <summary>Click to expand!</summary>

```
➜  Downloads deltachat --devmode
Logfile: /home/recolic/.config/DeltaChat/logs/2022-01-14-13-57-33.log
0s [i]main/index: Deltachat Version 1.26.0 tags/v1.26.0 1641982684699
0.1s [D]load-transaltions: { message: 'Select a chat or create a new chat' }
0.1s [i]main/index: cwd /home/recolic/.config/DeltaChat/accounts
0.1s [D]main/deltachat: Initiating DeltaChatNode
0.1s [D]main/deltachat: Starting event handler
0.1s [i]main/deltachat: Ready, starting accounts io...
0.1s [i]main/deltachat: Started accounts io.
0.2s [i]main/menu: rebuilding menu with locale en
0.2s [i]main/mainWindow: toggleDevTools
0.2s [D]logger/log-cleanup: Nothing to do (not more than 10 logfiles to delete)
[1499058:0114/135734.047634:ERROR:cert_issuer_source_aia.cc(32)] Error parsing cert retrieved from AIA (as DER):
ERROR: Couldn't read tbsCertificate as SEQUENCE
ERROR: Failed parsing Certificate

[1499058:0114/135734.049713:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for 0.0.0.0 failed:
----- Certificate i=0 (CN=recolicmspc.mscorp.rtmp.asia) -----
ERROR: Time is after notAfter


0.6s [D]main/ipc: EVENT_DC_DISPATCH_CB (0) : extras.getActiveTheme 
0.6s [D]main/deltachat/extras: theme system
0.6s [D]main/themes: load theme file system /opt/DeltaChat/electron_app/themes/light.css
0.6s [D]main/themes: render theme data
0.6s [D]main/themes: render theme data for theme: { name: 'Light Theme', description: 'Our default theme.' }
0.9s [D]main/ipc: EVENT_DC_DISPATCH_CB (1) : getState 
init: 898.581ms
0.9s [D]main/ipc: EVENT_DC_DISPATCH_CB (2) : extras.getLocaleData 
0.9s [D]main/ipc: EVENT_DC_DISPATCH_CB (3) : settings.getDesktopSettings 
0.9s [D]main/ipc: EVENT_DC_DISPATCH_CB (4) : login.getLastLoggedInAccount 
0.9s [D]main/ipc: EVENT_DC_DISPATCH_CB (5) : settings.getDesktopSettings 
0.9s [D]main/ipc: EVENT_DC_DISPATCH_CB (6) : login.getAllAccounts 
1.6s [i]main/state: Saving state to /home/recolic/.config/DeltaChat/config.json
6.5s [D]main/ipc: EVENT_DC_DISPATCH_CB (7) : login.addAccount 
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:23: First time init: creating tables
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:374: [migration] v69
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:386: [migration] v71
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:403: [migration] v72
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:415: [migration] v73
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:451: [migration] v74
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:456: [migration] v75
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:464: [migration] v76
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:469: [migration] v77
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:476: [migration] v78
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:481: [migration] v79
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql/migrations.rs:491: [migration] v80
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/sql.rs:201: Opened database Path { inner: "/home/recolic/.config/DeltaChat/accounts/441287d3eb0a4410bf7155c71609b8dc/dc.db" }.
6.5s [D]main/ipc: EVENT_DC_DISPATCH_CB (8) : login.selectAccount 1
6.5s [D]main/deltachat/login: selectAccount 1
6.5s [i]main/deltachat/login: Ready, starting io...
6.5s [D]main/deltachat/login: Started IO
6.5s [i]main/deltachat/login: dc_get_info {
  arch: '64',
  bcc_self: '0',
  blobdir: '/home/recolic/.config/DeltaChat/accounts/441287d3eb0a4410bf7155c71609b8dc/dc.db-blobs',
  bot: '0',
  configured_mvbox_folder: '<unset>',
  configured_sentbox_folder: '<unset>',
  database_dir: '/home/recolic/.config/DeltaChat/accounts/441287d3eb0a4410bf7155c71609b8dc/dc.db',
  database_version: '80',
  delete_device_after: '0',
  delete_server_after: '0',
  deltachat_core_version: 'v1.70.0',
  display_name: '0',
  download_limit: '0',
  e2ee_enabled: '1',
  entered_account_settings: 'unset imap:unset:0:unset:0:cert_automatic smtp:unset:0:unset:0:cert_automatic 0',
  fetch_existing_msgs: '1',
  fingerprint: '<key failure: No address configured>',
  folders_configured: '0',
  inbox_watch: '1',
  is_configured: '0',
  journal_mode: 'wal',
  key_gen_type: '0',
  last_housekeeping: '0',
  level: 'awesome',
  mdns_enabled: '1',
  media_quality: '0',
  messages_in_contact_requests: '0',
  mvbox_move: '1',
  mvbox_watch: '1',
  num_cpus: '8',
  number_of_chat_messages: '0',
  number_of_chats: '0',
  number_of_contacts: '0',
  private_key_count: '0',
  public_key_count: '0',
  quota_exceeding: '0',
  scan_all_folders_debounce_secs: '60',
  selfavatar: '<unset>',
  send_sync_msgs: '0',
  sentbox_move: '0',
  sentbox_watch: '1',
  show_emails: '0',
  socks5_enabled: '0',
  sqlite_version: '3.37.1',
  uptime: '0h 0m 0s',
  used_account_settings: 'unset imap:unset:0:unset:0:cert_automatic smtp:unset:0:unset:0:cert_automatic 0',
  webrtc_instance: '<unset>'
}
6.5s [i]core/event: 1 DC_EVENT_INFO 0 src/context.rs:168: starting IO
6.5s [D]main/deltachat: sendToRenderer eventType: DC_EVENT_INFO
6.5s [E]core/event: 1 DC_EVENT_ERROR 0 Failed to start IO: IMAP Connect without configured params
6.5s [D]main/deltachat: sendToRenderer eventType: DC_EVENT_ERROR
6.5s [D]core/event: 1 DC_EVENT_NEW_BLOB_FILE 0 $BLOBDIR/icon-device.png
6.5s [D]main/deltachat: sendToRenderer eventType: DC_EVENT_NEW_BLOB_FILE
6.5s [D]core/event: 1 DC_EVENT_MSGS_CHANGED 10 10
6.5s [D]main/deltachat: sendToRenderer eventType: DC_EVENT_MSGS_CHANGED
6.5s [D]main/deltachat: sendToRenderer eventType: DD_EVENT_CHATLIST_CHANGED
6.5s [w]core/event: 1 DC_EVENT_WARNING 0 deltachat-ffi/src/lib.rs:227: invalid stock message id 81
6.5s [D]main/deltachat: sendToRenderer eventType: DC_EVENT_WARNING
6.5s [D]main/ipc: EVENT_DC_DISPATCH_CB (9) : login.accountInfo 1
12.2s [D]main/ipc: EVENT_DC_DISPATCH_CB (10) : getProviderInfo root@rec
[Recolic note: the program hangs for 30 seconds now. The UI is not responding]
42.9s [D]main/ipc: EVENT_DC_DISPATCH_CB (11) : getProviderInfo root@recolic.netrootroot
```

</details>

Basing on the log, I suggest you to use `onfocusout=someHandler` instead of `onchange=someHandler`.

**Comment by Jikstra** [comment]
```
12.2s [D]main/ipc: EVENT_DC_DISPATCH_CB (10) : getProviderInfo root@rec
[Recolic note: the program hangs for 30 seconds now. The UI is not responding]
```

This is really weird. One single getProviderInfo is so slow that it hangs frontend&backend for 30seconds. I will probably also not help to change it to an `onfocusout=someHandler`as the 30sec hang would happen after you focus out, which would be also not good.

**Comment by recolic** [comment]
I believe I have found the root cause. There's something wrong with **our DNS server**. 

I tried to run `dig rec` or `drill rec` on other Linux machines, it immediately failed without answer.    
However, while running `drill rec` on my machine, the DNS server is not responding at all. The UDP packet is not responded, and current process is blocked by the network operation. That's why deltachat hangs for 30 seconds

I did some tests in different network condition, and different linux distributions. The problem is not reproduced in other networks at all. 

Attaching the wireshark capture result for reference: <https://recolic.cc/tmp/capt-filtered-validto20221231.pcapng>

![screenshot1](https://user-images.githubusercontent.com/19610256/150105250-9603e7e7-4f8f-4575-9d8b-550008590e8f.png)

![screenshot2](https://user-images.githubusercontent.com/19610256/150105327-77de08df-52d5-4930-8226-d76230cc9744.png)
